client-datastore-cassandra/src/main/java/com/telecominfraproject/wlan/client/datastore/cassandra/ClientSessionDAO.java

```diff
@@ -403,10 +431,29 @@ public ClientSession deleteSession(int customerId, long equipmentId, MacAddress
             strb.append(") ");
 
             query = strb.toString() + query;
-            query_head = "select macAddress from client_session_by_location where ";
+            query_head = "select macAddress from client_session_by_location where customerId = ? ";
             
             filterOptions = FilterOptions.customer_and_location;
         }
+        
+        // Add macSubstring filter
+        if (macSubstring != null) {
+        	// Rebuilding query args
+        	queryArgs.clear();
+        	queryArgs.add("%" + macSubstring.toLowerCase() + "%");
+        	
+        	if (locationIds != null && !locationIds.isEmpty()) {
+        		queryArgs.addAll(locationIds);
+        	}
+        	if (equipmentIds != null && !equipmentIds.isEmpty()) {
+        		queryArgs.addAll(equipmentIds);
+        	}
+        	
+        	query = CQL_APPEND_SEARCH_MAC_STRING + query;
+        	query_head = "select macAddress from client_session_by_mac_string where customerId = ? ";
```

Comment:
same here - the query_head is used together with the query, customerId bind value is not set, this should not work as it is written.